### ISSUE TITLE
Migrate to solc 0.8

### ISSUE THREAD

**Issue by nventuro** [balancer-labs/balancer-v2-monorepo#1340]
At some we'll want to do this.

There's two big changes that I know of that will affect us:
 - enums will become 256 bits wide, as opposed to only as big as needed (typically uint8). This affects some things, incl. the recovery mode enum decoding from #1332 - but I believe abi encoding and decoding 8 bit values is the exact same as 256 bit ones, in which case we'd be fine.
 - arithmetic is checked by default. Eventually we'll just delete `Math.add` and `Math.sub`, but meanwhile we can gradually replace their usage for the native operators.

**Comment by nventuro** [comment]
We may want to wait until https://github.com/ethereum/solidity/issues/10698 is solved before migrating. See [this comment](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3431#issuecomment-1144104066) on how the lack of that feature negatively affects gas usage.

**Comment by nventuro** [comment]
After reading the docs more carefully, I realized that the actual change 0.8 introduced is to make enums always have at most 8 elements, not make the type wider. So we're unaffected by this.

**Comment by Starloss** [comment]
Any update on this?

**Comment by nventuro** [comment]
Not really. We still want to do it, and I was hoping the `unchecked` expression would be resolved by now. Unfortunately it looks like it's not going to happen, so we may just have to bite the bullet and live with that.

**Comment by 0xJem** [comment]
Hello! Are there still plans to perform this migration? We (Olympus) have a codebase built on solc 0.8.x and were trying to use the StableMath library, but are concerned about changes in behaviour (particularly those not immediately visible, edge cases, etc) due to the overflow/underflow changes.
